src/macros.rs

```diff
@@ -239,103 +255,117 @@ fn rewrite_macro_inner(
     } = match parse_macro_args(context, ts, style, is_forced_bracket) {
         Some(args) => args,
         None => {
-            return return_macro_parse_failure_fallback(
-                context,
-                shape.indent,
-                position,
-                mac.span(),
+            return (
+                return_macro_parse_failure_fallback(context, shape.indent, position, mac.span()),
+                None,
             );
         }
     };
 
     if !arg_vec.is_empty() && arg_vec.iter().all(MacroArg::is_item) {
-        return rewrite_macro_with_items(
-            context,
-            &arg_vec,
-            &macro_name,
-            shape,
-            style,
-            original_style,
-            position,
-            mac.span(),
+        return (
+            rewrite_macro_with_items(
+                context,
+                &arg_vec,
+                &macro_name,
+                shape,
+                style,
+                original_style,
+                position,
+                mac.span(),
+            ),
+            if original_style != style {
+                Some(style)
+            } else {
+                None
+            },
         );
     }
 
-    match style {
-        Delimiter::Parenthesis => {
-            // Handle special case: `vec!(expr; expr)`
-            if vec_with_semi {
-                handle_vec_semi(context, shape, arg_vec, macro_name, style)
-            } else {
-                // Format macro invocation as function call, preserve the trailing
-                // comma because not all macros support them.
-                overflow::rewrite_with_parens(
-                    context,
-                    &macro_name,
-                    arg_vec.iter(),
-                    shape,
-                    mac.span(),
-                    context.config.fn_call_width(),
-                    if trailing_comma {
+    (
+        match style {
+            Delimiter::Parenthesis => {
+                // Handle special case: `vec!(expr; expr)`
+                if vec_with_semi {
+                    handle_vec_semi(context, shape, arg_vec, macro_name, style)
```

Comment:
Instead of adding another level of indentation by wrapping this all in 1 big tuple, it's easier to assign the result of the match to a variable. Then we can just return the the `(rewrite, Some(style))`

```rust
let rewrite = match style { /* code goes here */};
return (rewrite, Some(style));